### ISSUE TITLE
`==` doesn't work for `SubStaticString`

### ISSUE THREAD

**Issue by nhz2** [mkitti/StaticStrings.jl#14]
```julia
julia> using StaticStrings

julia> a = SubStaticString("hello",1:2)
substatic"he"5

julia> b = SubStaticString("hello",3:4)
substatic"ll"5

julia> a == b
true
```

Also, is there a difference between `SubStaticString{N}` and just `SubString{StaticString{N}}`?

**Comment by mkitti** [comment]
This is a bug. There is a difference in that `SubStaticString` is itself an `AbstractStaticString`.

**Comment by mkitti** [comment]
Fixed in #16, 0.2.4

**Comment by mkitti** [comment]
Actually fixed in #17, 0.2.5

```julia
julia> using StaticStrings

julia> a = SubStaticString("hello",1:2)
substatic"he"5

julia> b = SubStaticString("hello",3:4)
substatic"ll"5

julia> a == b
false
```
